Model/Sync/Catalog/Processor.php

```diff
@@ -707,9 +706,16 @@ protected function existDataRequest($url, $requestIds, $sqlData, $type = 'produc
      */
     protected function getProductsForCategorySync($data, $collectionItems, array $dataForCategorySync): array
     {
+        $storeId = $this->coreConfig->getStoreId();
+        $retryItems = [];
+        if ($this->retryItems && isset($this->retryItems[$storeId])) {
```

Comment:
I see _four_not_four_data_ is defining `$this->retryItems`, how does that work?

Reply:
This is the method, where it decides whether the product is mapped to some categories and should it need to call the category sync process. If $this->retryItems contains something , that means the product in the current iteration is not successfully synced to Yotpo (it was either 404 or 409 response) and this product is added to the retryItems array. At this time it will not have a valid yotpo_id and we should not call the category sync process, unless those retryItems are resent to Yotpo with apropriate sync methods (a Create OR a Patch) . This retry is being tried in the same iteration later(Lines 427 to 429)